### PR TITLE
Homework6

### PR DESCRIPTION
**Pull request by akuzmichova**
Hi Deepak, can you please review

### CODE REVIEW

File: src/test/java/Pages/Darksky/Homepage.java [hunk 1]
```diff
@@ -65,6 +72,39 @@ public void clickTMButton() {
...
+    public List<String> getTimeDataPointsFromTimeline() {
+        for (WebElement element : getElements(timeDataPoints)) {
+            timeDataPointsFromTimeline.add(getElementText(element));
+        }
+        return timeDataPointsFromTimeline;
+    }
+
+    List<String> timeDataPointsFromCalendar = new ArrayList<>();
+
+    public List<String> getTimeLineFromCalendar() {
```
Comment (deepaktechnosoft) [inline src/test/java/Pages/Darksky/Homepage.java:93]: Good code. My suggestion would be to move it in the Utils class.
As it is not an action or data user getting from WebPage.

File: src/test/java/Pages/Facebook/HomepageFB.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import Pages.WebCommands;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import java.util.List;
+
+public class HomepageFB extends WebCommands {
+
+    By createNewAccLocator = By.linkText("Create New Account");
+    By genderRadioButtons = By.className("_8esa");
```
Comment (deepaktechnosoft) [inline src/test/java/Pages/Facebook/HomepageFB.java:11]: use better locator, class name looks random data

File: src/test/java/Pages/Facebook/HomepageFB.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    public List<WebElement> getGenderRadioButtons() {
+        return getElements(genderRadioButtons);
+    }
+
+    public boolean areGenderRadioButtonsSelected() {
+        boolean isGenderRadioButtonSelected = false;
+        for (WebElement genderRadioButton : getGenderRadioButtons()) {
+            isGenderRadioButtonSelected &= isElementSelected(genderRadioButton);
```
Comment (deepaktechnosoft) [inline src/test/java/Pages/Facebook/HomepageFB.java:25]: I think, you should use OR operator instead AND.
bcz, even if all radio buttons are selected, you will have 'false' in isGenderRadioButtonSelected

File: src/test/java/StepDefinition/Facebook/MessengerSD.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    @And("On Messenger page I click on Log In button")
+    public void clickMessengerLogInButton() {
+        mes.switchToMessengerHandle();
+        mes.clickMessengerLogInButton();
+    }
+
+    @Then("I verify error message \"Incorrect email or phone number\" is displayed")
+    public void verifyErrorMessageDisplayed() {
+        mes.switchToMessengerLoginHandle();
```
Comment (deepaktechnosoft) [inline src/test/java/StepDefinition/Facebook/MessengerSD.java:29]: does it open a new window??

**Review by deepaktechnosoft — CHANGES_REQUESTED** [review]

File: src/test/java/StepDefinition/Facebook/MessengerSD.java (hunk 4 above)
Comment (akuzmichova) [reply src/test/java/StepDefinition/Facebook/MessengerSD.java:29]: Hi, yes, it is opening new window

File: src/test/java/Pages/Facebook/HomepageFB.java (hunk 3 above)
Comment (akuzmichova) [reply src/test/java/Pages/Facebook/HomepageFB.java:25]: Agree

File: src/test/java/Pages/Facebook/HomepageFB.java (hunk 2 above)
Comment (akuzmichova) [reply src/test/java/Pages/Facebook/HomepageFB.java:11]: changed to By.xpath("//input[@type='radio']");

**Review by deepaktechnosoft — APPROVED** [review]

**Pull request merged**